service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -201,11 +202,28 @@ private void addOperation(Operation operation) {
     }
   }
 
+  public void updateQueryTag(String queryId, String queryTag) {
+    Operation operation = queryIdOperation.get(queryId);
+    if (operation != null) {
+      if (queryTagToIdMap.get(queryTag) != null) {
+        throw new RuntimeException("tag " + queryTag + " is already applied for query " + queryId);
```

Comment:
This log msg is incorrect. queryId should be from the map not the input.

Reply:
done